<h1>Palmer Penguin Data Analytics</h1>
<hr>
<h3>Analyst: John Kevin Aguilera</h3>

In [1]:
import numpy as np
import pandas as pd

<p> Importing datasets to Jupyter notebook</p>
<p> Converting datasets to dataframe </p>
<ul>Requirements
    <li>Name of the dataframe</li>
    <li>Source</li>

In [2]:
penguins = pd.read_csv('datasets\\penguins.csv')

In [3]:
penguins

,species,island,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g,sex,year
0,Adelie,Torgersen,39.1,18.7,181.0,3750.0,male,2007
1,Adelie,Torgersen,39.5,17.4,186.0,3800.0,female,2007
2,Adelie,Torgersen,40.3,18.0,195.0,3250.0,female,2007
3,Adelie,Torgersen,NaN,NaN,NaN,NaN,NaN,2007
4,Adelie,Torgersen,36.7,19.3,193.0,3450.0,female,2007
...,...,...,...,...,...,...,...,...
339,Chinstrap,Dream,55.8,19.8,207.0,4000.0,male,2009
340,Chinstrap,Dream,43.5,18.1,202.0,3400.0,female,2009
341,Chinstrap,Dream,49.6,18.2,193.0,3775.0,male,2009
342,Chinstrap,Dream,50.8,19.0,210.0,4100.0,male,2009


<h1>How to fillup Null values</h1>

In [4]:
chinstrap_weight = penguins[(penguins['species'] == "Chinstrap") & (penguins['body_mass_g'] >= 4000)]

<h5>Filling up bill_lenght_mm</h5>

In [5]:
penguins['bill_length_mm'].mean()

np.float64(43.9219298245614)

In [6]:
penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

0      39.10000
1      39.50000
2      40.30000
3      43.92193
4      36.70000
         ...   
339    55.80000
340    43.50000
341    49.60000
342    50.80000
343    50.20000
Name: bill_length_mm, Length: 344, dtype: float64

In [7]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

In [8]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<p>Filling up sex</p>

In [9]:
penguins['sex'].mode()

0    male
Name: sex, dtype: object

In [10]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [11]:
penguins['bill_depth_mm'].mean()

np.float64(17.151169590643274)

In [12]:
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())

In [13]:
penguins['flipper_length_mm'].mean()

np.float64(200.91520467836258)

In [14]:
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())

In [15]:
penguins['body_mass_g'].mean()

np.float64(4201.754385964912)

In [16]:
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

In [17]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h1>Descriptive Analysis</h1> 

In [18]:
penguins.describe()

,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g,year
count,344.000000,344.000000,344.000000,344.000000,344.000000
mean,43.921930,17.151170,200.915205,4201.754386,2008.029070
std,5.443643,1.969027,14.020657,799.613058,0.818356
min,32.100000,13.100000,172.000000,2700.000000,2007.000000
25%,39.275000,15.600000,190.000000,3550.000000,2007.000000
50%,44.250000,17.300000,197.000000,4050.000000,2008.000000
75%,48.500000,18.700000,213.000000,4750.000000,2009.000000
max,59.600000,21.500000,231.000000,6300.000000,2009.000000


In [19]:
penguins.describe().T

,count,mean,std,min,25%,50%,75%,max
bill_length_mm,344.0,43.921930,5.443643,32.1,39.275,44.25,48.5,59.6
bill_depth_mm,344.0,17.151170,1.969027,13.1,15.600,17.30,18.7,21.5
flipper_length_mm,344.0,200.915205,14.020657,172.0,190.000,197.00,213.0,231.0
body_mass_g,344.0,4201.754386,799.613058,2700.0,3550.000,4050.00,4750.0,6300.0
year,344.0,2008.029070,0.818356,2007.0,2007.000,2008.00,2009.0,2009.0


<h3>Cleaning the dataframe</h3>

In [20]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h2>Selecting the column</h2>
<h3>Syntax</h3><br>
data_frame["column_name"]

<h1>Creating dataframes</h1>

In [21]:
penguins['species'].unique()

array(['Adelie', 'Gentoo', 'Chinstrap'], dtype=object)

In [22]:
adelie_df = penguins[penguins['species'] == 'Adelie']

In [23]:
adelie_df

,species,island,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g,sex,year
0,Adelie,Torgersen,39.10000,18.70000,181.000000,3750.000000,male,2007
1,Adelie,Torgersen,39.50000,17.40000,186.000000,3800.000000,female,2007
2,Adelie,Torgersen,40.30000,18.00000,195.000000,3250.000000,female,2007
3,Adelie,Torgersen,43.92193,17.15117,200.915205,4201.754386,male,2007
4,Adelie,Torgersen,36.70000,19.30000,193.000000,3450.000000,female,2007
...,...,...,...,...,...,...,...,...
147,Adelie,Dream,36.60000,18.40000,184.000000,3475.000000,female,2009
148,Adelie,Dream,36.00000,17.80000,195.000000,3450.000000,female,2009
149,Adelie,Dream,37.80000,18.10000,193.000000,3750.000000,male,2009
150,Adelie,Dream,36.00000,17.10000,187.000000,3700.000000,female,2009


In [24]:
gentoo_df = penguins[penguins['species'] == "Gentoo"]

In [25]:
gentoo_df

,species,island,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g,sex,year
152,Gentoo,Biscoe,46.10000,13.20000,211.000000,4500.000000,female,2007
153,Gentoo,Biscoe,50.00000,16.30000,230.000000,5700.000000,male,2007
154,Gentoo,Biscoe,48.70000,14.10000,210.000000,4450.000000,female,2007
155,Gentoo,Biscoe,50.00000,15.20000,218.000000,5700.000000,male,2007
156,Gentoo,Biscoe,47.60000,14.50000,215.000000,5400.000000,male,2007
...,...,...,...,...,...,...,...,...
271,Gentoo,Biscoe,43.92193,17.15117,200.915205,4201.754386,male,2009
272,Gentoo,Biscoe,46.80000,14.30000,215.000000,4850.000000,female,2009
273,Gentoo,Biscoe,50.40000,15.70000,222.000000,5750.000000,male,2009
274,Gentoo,Biscoe,45.20000,14.80000,212.000000,5200.000000,female,2009


In [26]:
chinstrap_df = penguins[penguins['species'] == "Chinstrap"]

In [27]:
chinstrap_df

,species,island,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g,sex,year
276,Chinstrap,Dream,46.5,17.9,192.0,3500.0,female,2007
277,Chinstrap,Dream,50.0,19.5,196.0,3900.0,male,2007
278,Chinstrap,Dream,51.3,19.2,193.0,3650.0,male,2007
279,Chinstrap,Dream,45.4,18.7,188.0,3525.0,female,2007
280,Chinstrap,Dream,52.7,19.8,197.0,3725.0,male,2007
...,...,...,...,...,...,...,...,...
339,Chinstrap,Dream,55.8,19.8,207.0,4000.0,male,2009
340,Chinstrap,Dream,43.5,18.1,202.0,3400.0,female,2009
341,Chinstrap,Dream,49.6,18.2,193.0,3775.0,male,2009
342,Chinstrap,Dream,50.8,19.0,210.0,4100.0,male,2009


In [28]:
penguins['body_mass_g'] >= 4000

0      False
1      False
2      False
3       True
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Name: body_mass_g, Length: 344, dtype: bool

In [29]:
chinstrap_weight.info()

<class 'pandas.core.frame.DataFrame'>
Index: 16 entries, 284 to 342
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            16 non-null     object 
 1   island             16 non-null     object 
 2   bill_length_mm     16 non-null     float64
 3   bill_depth_mm      16 non-null     float64
 4   flipper_length_mm  16 non-null     float64
 5   body_mass_g        16 non-null     float64
 6   sex                16 non-null     object 
 7   year               16 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 1.1+ KB


<h1>How to change data type of a column</h1>

In [30]:
penguins['species'] = penguins['species'].astype('category')
penguins['year'] = penguins['year'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')

In [31]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


In [32]:
penguins.describe()

,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g
count,344.000000,344.000000,344.000000,344.000000
mean,43.921930,17.151170,200.915205,4201.754386
std,5.443643,1.969027,14.020657,799.613058
min,32.100000,13.100000,172.000000,2700.000000
25%,39.275000,15.600000,190.000000,3550.000000
50%,44.250000,17.300000,197.000000,4050.000000
75%,48.500000,18.700000,213.000000,4750.000000
max,59.600000,21.500000,231.000000,6300.000000


In [33]:
penguins['mass_kg'] = np.round(penguins['body_mass_g'] / 1000,1)

In [34]:
penguins

,species,island,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g,sex,year,mass_kg
0,Adelie,Torgersen,39.10000,18.70000,181.000000,3750.000000,male,2007,3.8
1,Adelie,Torgersen,39.50000,17.40000,186.000000,3800.000000,female,2007,3.8
2,Adelie,Torgersen,40.30000,18.00000,195.000000,3250.000000,female,2007,3.2
3,Adelie,Torgersen,43.92193,17.15117,200.915205,4201.754386,male,2007,4.2
4,Adelie,Torgersen,36.70000,19.30000,193.000000,3450.000000,female,2007,3.4
...,...,...,...,...,...,...,...,...,...
339,Chinstrap,Dream,55.80000,19.80000,207.000000,4000.000000,male,2009,4.0
340,Chinstrap,Dream,43.50000,18.10000,202.000000,3400.000000,female,2009,3.4
341,Chinstrap,Dream,49.60000,18.20000,193.000000,3775.000000,male,2009,3.8
342,Chinstrap,Dream,50.80000,19.00000,210.000000,4100.000000,male,2009,4.1


<h1>Exploratory Data Analytics (EDA)</h1>

<h1>1. Which island has the most penguins?</h1>

In [35]:
penguins['island'].unique()

['Torgersen', 'Biscoe', 'Dream']
Categories (3, object): ['Biscoe', 'Dream', 'Torgersen']

In [36]:
torgersen_island = penguins['island'] == 'Torgersen'

In [37]:
torgersen_island.value_counts()

island
False    292
True      52
Name: count, dtype: int64

In [38]:
biscoe_island = penguins['island'] == 'Biscoe'

In [39]:
biscoe_island.value_counts()

island
False    176
True     168
Name: count, dtype: int64

In [40]:
dream_island = penguins['island'] == 'Dream'

In [41]:
dream_island.value_counts()

island
False    220
True     124
Name: count, dtype: int64

<h3>Insight #1: The island that has the most penguins recorded is BISCOE.</h3>

<h1>2. Which year has the most recorded penguins with the body mass of over 3500g? </h1>

In [42]:
penguins['year'].unique()

[2007, 2008, 2009]
Categories (3, int64): [2007, 2008, 2009]

In [43]:
year_2007_df = penguins[penguins['year'] == 2007] 

In [125]:
bdm_2007 = year_2007_df['body_mass_g'] >= 3500

In [126]:
bdm_2007.value_counts()

body_mass_g
True     84
False    26
Name: count, dtype: int64

In [46]:
year_2008_df = penguins[penguins['year'] == 2008]

In [128]:
bdm_2008 = year_2008_df['body_mass_g'] >= 3500

In [129]:
bdm_2008.value_counts()

body_mass_g
True     98
False    16
Name: count, dtype: int64

In [49]:
year_2009_df = penguins[penguins['year'] == 2009]

In [130]:
bdm_2009 = year_2009_df['body_mass_g'] >= 3500

In [131]:
bdm_2009.value_counts()

body_mass_g
True     91
False    29
Name: count, dtype: int64

<h3>Insight #2: The year 2008 recorded the highest number of penguins with a body mass above 3500g</h3>

<h1>3. Which species has the longest bill length, deepest bill depth, and heaviest body mass?</h1>

In [53]:
adelie_bl = adelie_df['bill_length_mm'].mean()

In [54]:
adelie_bl

np.float64(38.82514427516159)

In [55]:
gentoo_bl = gentoo_df['bill_length_mm'].mean()

In [56]:
gentoo_bl

np.float64(47.475983305036785)

In [57]:
chinstrap_bl = chinstrap_df['bill_length_mm'].mean()

In [58]:
chinstrap_bl

np.float64(48.83382352941177)

In [59]:
adelie_bd = adelie_df['bill_depth_mm'].mean()

In [60]:
adelie_bd

np.float64(18.338494536780548)

In [61]:
gentoo_bd = gentoo_df['bill_depth_mm'].mean()

In [62]:
gentoo_bd

np.float64(14.999606206376153)

In [63]:
chinstrap_bd = chinstrap_df['bill_depth_mm'].mean()

In [64]:
chinstrap_bd

np.float64(18.42058823529412)

In [65]:
adelie_bm = adelie_df['body_mass_g'].mean()

In [66]:
adelie_bm

np.float64(3703.95891043398)

In [67]:
gentoo_bm = gentoo_df['body_mass_g'].mean()

In [68]:
gentoo_bm

np.float64(5068.965761177136)

In [69]:
chinstrap_bm = chinstrap_df['body_mass_g'].mean()

In [70]:
chinstrap_bm

np.float64(3733.0882352941176)

<h5>Evaluating the data using logical operators</h5>

In [71]:
adelie_bl > gentoo_bl

np.False_

In [72]:
gentoo_bl > chinstrap_bl

np.False_

In [73]:
chinstrap_bl > adelie_bl

np.True_

In [74]:
chinstrap_bl > adelie_bl or gentoo_bl

np.True_

<p>Chinstrap has the longest bill length out of all species recorded</p>

In [75]:
adelie_bd > gentoo_bd

np.True_

In [76]:
gentoo_bd > chinstrap_bd

np.False_

In [77]:
chinstrap_bd > adelie_bd

np.True_

In [78]:
chinstrap_bd > adelie_bd or gentoo_bd

np.True_

<p>Chinstrap has the deepest bill depth out of all species recorded</p>

In [79]:
adelie_bm > gentoo_bm

np.False_

In [80]:
gentoo_bm > chinstrap_bm

np.True_

In [81]:
chinstrap_bm > adelie_bm

np.True_

In [82]:
gentoo_bm > adelie_bm or chinstrap_bm

np.True_

<p>Gentoo has the heaviest body mass out of all species recorded</p>

<h3>Insight #3: Chinstrap had the longest bill depth and deepest bill depth, while Gentoo had the heaviest body mass.</h3>

<h1>Question #4: What is the demographic of Torgersen Island, and which year had the highest number of Adelie, Gentoo, and Chinstrap species recorded? </h1>

In [83]:
penguins['island'].unique()

['Torgersen', 'Biscoe', 'Dream']
Categories (3, object): ['Biscoe', 'Dream', 'Torgersen']

In [84]:
torgersen_df = penguins[penguins['island'] == 'Torgersen']

In [85]:
torgersen_adelie = torgersen_df['species'] == 'Adelie'

In [86]:
torgersen_gentoo = torgersen_df['species'] == 'Gentoo'

In [87]:
torgersen_chinstrap = torgersen_df['species'] == 'Chinstrap'

In [88]:
torgersen_adelie.value_counts()

species
True    52
Name: count, dtype: int64

In [89]:
torgersen_gentoo.value_counts()

species
False    52
Name: count, dtype: int64

In [90]:
torgersen_chinstrap.value_counts()

species
False    52
Name: count, dtype: int64

In [91]:
torgersen_2007 = torgersen_df['year'] == 2007
torgersen_2008 = torgersen_df['year'] == 2008
torgersen_2009 = torgersen_df['year'] == 2009

In [92]:
torgersen_2007.value_counts()

year
False    32
True     20
Name: count, dtype: int64

In [93]:
torgersen_2008.value_counts()

year
False    36
True     16
Name: count, dtype: int64

In [94]:
torgersen_2009.value_counts()

year
False    36
True     16
Name: count, dtype: int64

<h3>Insight #4: The island Torgersen only had Adelie as its inhabitant, with 2007 being the highest number of Adelies recorded.</h3>

<h1>Question #5: What is the demography of Island Biscoe, and which year had the highest number of Adelie, Gentoo, and Chinstrap species recorded? </h1>

In [95]:
penguins['island'].unique()

['Torgersen', 'Biscoe', 'Dream']
Categories (3, object): ['Biscoe', 'Dream', 'Torgersen']

In [96]:
biscoe_df = penguins[penguins['island'] == 'Biscoe']

In [97]:
biscoe_adelie = biscoe_df['species'] == 'Adelie'
biscoe_gentoo = biscoe_df['species'] == 'Gentoo'
biscoe_chinstrap = biscoe_df['species'] == 'Chinstrap'

In [98]:
biscoe_adelie.value_counts()

species
False    124
True      44
Name: count, dtype: int64

In [99]:
biscoe_gentoo.value_counts()

species
True     124
False     44
Name: count, dtype: int64

In [100]:
biscoe_chinstrap.value_counts()

species
False    168
Name: count, dtype: int64

In [101]:
biscoe_2007_adelie = (biscoe_df['year'] == 2007) & (biscoe_df['species'] == 'Adelie')
biscoe_2007_gentoo = (biscoe_df['year'] == 2007) & (biscoe_df['species'] == 'Gentoo')

biscoe_2008_adelie = (biscoe_df['year'] == 2008) & (biscoe_df['species'] == 'Adelie')
biscoe_2008_gentoo = (biscoe_df['year'] == 2008) & (biscoe_df['species'] == 'Gentoo')

biscoe_2009_adelie = (biscoe_df['year'] == 2009) & (biscoe_df['species'] == 'Adelie')
biscoe_2009_gentoo = (biscoe_df['year'] == 2009) & (biscoe_df['species'] == 'Gentoo')

In [102]:
biscoe_2007_adelie.value_counts()

False    158
True      10
Name: count, dtype: int64

In [103]:
biscoe_2007_gentoo.value_counts()

False    134
True      34
Name: count, dtype: int64

In [104]:
biscoe_2008_adelie.value_counts()

False    150
True      18
Name: count, dtype: int64

In [105]:
biscoe_2008_gentoo.value_counts()

False    122
True      46
Name: count, dtype: int64

In [106]:
biscoe_2009_adelie.value_counts()

False    152
True      16
Name: count, dtype: int64

In [107]:
biscoe_2009_gentoo.value_counts()

False    124
True      44
Name: count, dtype: int64

<h3>Insight #5: The island Biscoe had 44 Adelie and 124 Gentoo as its inhabitants, with the year 2008 recording 18 Adelies and 2009 recording 44 Gentoos.</h3>

<h1>Question #6: What is the demography of Island Dream, and which year had the highest number of Adelie, Gentoo, and Chinstrap species recorded? </h1>

In [108]:
penguins['island'].unique()

['Torgersen', 'Biscoe', 'Dream']
Categories (3, object): ['Biscoe', 'Dream', 'Torgersen']

In [109]:
dream_df = penguins[penguins['island'] == 'Dream']

In [110]:
dream_adelie = dream_df['species'] == 'Adelie'
dream_gentoo = dream_df['species'] == 'Gentoo'
dream_chinstrap = dream_df['species'] == 'Chinstrap'

In [111]:
dream_adelie.value_counts()

species
False    68
True     56
Name: count, dtype: int64

In [112]:
dream_gentoo.value_counts()

species
False    124
Name: count, dtype: int64

In [113]:
dream_chinstrap.value_counts()

species
True     68
False    56
Name: count, dtype: int64

In [114]:
dream_2007_adelie = (dream_df['year'] == 2007) & (dream_df['species'] == 'Adelie')
dream_2007_chinstrap = (dream_df['year'] == 2007) & (dream_df['species'] == 'Chinstrap')

dream_2008_adelie = (dream_df['year'] == 2008) & (dream_df['species'] == 'Adelie')
dream_2008_chinstrap = (dream_df['year'] == 2008) & (dream_df['species'] == 'Chinstrap')

dream_2009_adelie = (dream_df['year'] == 2009) & (dream_df['species'] == 'Adelie')
dream_2009_chinstrap = (dream_df['year'] == 2009) & (dream_df['species'] == 'Chinstrap')

In [115]:
dream_2007_adelie.value_counts()

False    104
True      20
Name: count, dtype: int64

In [116]:
dream_2007_chinstrap.value_counts()

False    98
True     26
Name: count, dtype: int64

In [117]:
dream_2008_adelie.value_counts()

False    108
True      16
Name: count, dtype: int64

In [118]:
dream_2008_chinstrap.value_counts()

False    106
True      18
Name: count, dtype: int64

In [119]:
dream_2009_adelie.value_counts()

False    104
True      20
Name: count, dtype: int64

In [120]:
dream_2009_chinstrap.value_counts()

False    100
True      24
Name: count, dtype: int64

<h3>Insight #6: The island Dream has 56 Adelie and 68 Chinstrap penguins as its inhabitants, with the year 2007 recording the highest number of Adelie and Chinstrap penguins.</h3>

<h1>Question #7: Which species has the highest number of male and female?</h1>

In [121]:
adelie_df['sex'].value_counts()

sex
male      79
female    73
Name: count, dtype: int64

In [122]:
gentoo_df['sex'].value_counts()

sex
male      66
female    58
Name: count, dtype: int64

In [123]:
chinstrap_df['sex'].value_counts()

sex
female    34
male      34
Name: count, dtype: int64

<h3>Insight #7: The species Adelie recorded the highest number of male and female with 79 and 73 respectively.</h3>